src/Symfony/Bundle/FrameworkBundle/Tests/Routing/DelegatingLoaderTest.php

```diff
@@ -0,0 +1,46 @@
+<?php
+
+namespace Symfony\Bundle\FrameworkBundle\Tests\Routing;
+
+use Psr\Log\NullLogger;
+use Symfony\Bundle\FrameworkBundle\Controller\ControllerNameParser;
+use Symfony\Bundle\FrameworkBundle\Routing\DelegatingLoader;
+use Symfony\Component\Config\Loader\LoaderResolver;
+
+class DelegatingLoaderTest extends \PHPUnit_Framework_TestCase
+{
+    /** @var ControllerNameParser */
+    private $controllerNameParser;
+
+    public function setUp()
+    {
+        $this->controllerNameParser = $this->getMockBuilder('Symfony\Bundle\FrameworkBundle\Controller\ControllerNameParser')
+            ->disableOriginalConstructor()
+            ->getMock();
+    }
+
+    /**
+     * @group legacy
+     */
+    public function testLegacyConstructorApi()
```

Comment:
please add another legacy test for the other possibility of the old constructor:

```php
new DelegatingLoader($this->controllerNameParser, null, new LoaderResolver());
```